test/unit/math/prim/err/elementwise_check_test.cpp

```diff
@@ -0,0 +1,149 @@
+#include <test/unit/util.hpp>
+#include <stan/math/prim.hpp>
+#include <stan/math/prim/fun/is_nan.hpp>
+#include <gtest/gtest.h>
+#include <limits>
+#include <vector>
+
+template <typename T>
+bool example_predicate(const T& x) {
+  return !stan::math::is_nan(x);
+}
+
```

Comment:
[optional]
This'd be easier just assigning a functor

```
auto p = [](const auto& x) { return !stan::math::is_nan(x); }
```

and then using it in the checks.

Reply:
Yes it would. Thanks.